classes/ActionScheduler_AdminView.php

```diff
@@ -35,9 +35,42 @@ public function init() {
 			} else {
 				add_action( 'admin_menu', array( $self, 'register_menu' ) );
 			}
+			add_action( 'admin_notices', array( $self, 'check_past_due_actions' ) );
 		}
 	}
 
+	/**
+	 * Check if there are any past due action
+	 *
+	 * This method will check if there are past due. Because Action Scheduler is action agnostic,
+	 * it will execute an action with a list of the actions.
+	 *
+	 * The action owners should be listening to this action, and do something to warn the users
+	 * about how to fix their WP-Cron or to run their actions manually.
+	 *
+	 * This method is executed by the `admin_notices`, so it is safe to render the admin notification
+	 * from the `action_scheduler_past_due_action` action.
+	 *
+	 */
+	public function check_past_due_actions() {
+		$instance = ActionScheduler_Store::instance();
+		if ( $cache = get_transient( __METHOD__ ) ) {
```

Comment:
Good idea to cache it so it repeatedly triggers the action without repeating the query :+1:

However, what happens if the actions are run? The cache value isn't being invalidated, so it will continue to be passed to callbacks on `'action_scheduler_past_due_action'` and seen as "pending" by those callbacks incorrectly until the transient expires.

The transient needs to be invalidated when an action is run. Ideally in a very efficient and scalable way.